codec/reflectcodec/type_codec.go

```diff
@@ -282,10 +309,14 @@ func (c *genericCodec) MarshalInto(value interface{}, p *wrappers.Packer) error
 	return c.marshal(reflect.ValueOf(value), p, c.maxSliceLen)
 }
 
+func (c *genericCodec) marshal(value reflect.Value, p *wrappers.Packer, maxSliceLen uint32) error {
+	return c.marshalWithOmitEmpty(value, p, maxSliceLen, false)
+}
+
 // marshal writes the byte representation of [value] to [p]
```

Comment:
could we update the comment to refer to `marshalWithOmitEmpty` and also update the semantics? (or move the comment back to `marshal`?)